pkg/scheduler/testing/wrappers.go

```diff
@@ -390,3 +390,18 @@ func (n *NodeWrapper) Label(k, v string) *NodeWrapper {
 	n.Labels[k] = v
 	return n
 }
+
+func (n *NodeWrapper) Unschedulable(v bool) *NodeWrapper {
+	n.Spec.Unschedulable = v
+	return n
+}
+
+func (n *NodeWrapper) Capacity(capacity v1.ResourceList) *NodeWrapper {
```

Comment:
I'd suggest using paired key-value string (or just values) instead of v1.ResourceList here. v1.ResourceList is too low-level API, which is friendly for new-users to build, and if you look at the usage of createNode(), there is only one caller specifying a non-nil resourceList.

**Option 1:**

```go
func (n *NodeWrapper) Capacity(pairs ...string) *NodeWrapper {
	res := v1.ResourceList{}
	for i := 0; i < len(pairs); i += 2 {
		k, v := pairs[i], pairs[i+1]
		res[v1.ResourceName(k)] = resource.MustParse(v)
	}
	n.Status.Capacity, n.Status.Allocatable = res, res
	return n
}
```

**Option 2:**

```go
func (n *NodeWrapper) Capacity(cpu, mem string, pods int64) *NodeWrapper {
	res := v1.ResourceList{
		v1.ResourcePods: *resource.NewQuantity(pods, resource.DecimalSI),
	}
	if len(cpu) != 0 {
		res[v1.ResourceCPU] = resource.MustParse(cpu)
	}
	if len(mem) != 0 {
		res[v1.ResourceMemory] = resource.MustParse(mem)
	}
	n.Status.Capacity, n.Status.Allocatable = res, res
	return n
}
```

Reply:
I would prefer Option 1, but accepting map[string]string, instead of the pairs of lists

Reply:
yup, option 1 is also my preference.

Reply:
nit: `map[v1.ResourceName]string` should be less error-prone.

Reply:
Good idea, I will make the change!